core/tests/test_utils.py

```diff
@@ -453,10 +453,18 @@ def mock_get_platform_parameter_value(
         Returns:
             PlatformDataTypes. The value of the platform parameter if the
             parameter is present in the platform_parameter_names list.
+
+        Raises:
+            Exception. If the parameter_name is not present in the
+                platform_parameter_names list.
         """
         platform_parameter_name_value_dict = dict(
             (x.value, y) for x, y in platform_parameter_name_value_tuples
         )
+        if parameter_name not in platform_parameter_name_value_dict:
+            raise Exception(
+                'Unknown platform parameter name: %s' % parameter_name
```

Comment:
Can we make this more explicit -- something like "The value for the platform parameter XYZ was needed in this test, but not specified in the XXX decorator. Please add a XXX decorator with this information."

Also just to confirm -- if a backend test uses a platform parameter (directly or indirectly), then they *must* specify its value, is that correct?

Reply:
Done.

Regarding the second question, this is true if the test uses the decorator i.e. if a test uses the decorator, it will need to exhaustively define all the platform parameter that it requires or it will throw the error above. There may still be tests that rely on calling the actual procedure (not mocking) and/or there may be other tests that mocks the method a different way (e.g. always return xxx).

Reply:
Thanks @kevintab95, sg. One question though -- have we modified all the tests that rely on calling the actual procedure / mock the method a different way to use the new decorator? I think that's important because, otherwise, devs will simply copy those old tests if it's the first example they see.

Reply:
The way I went about these changes were to find out places where the email related platform parameter is used and then modify those tests to use the decorator instead. I do see around 24 places in the test files where `platform_parameter_services.get_platform_parameter_value` is mocked without the decorator e.g.

https://github.com/oppia/oppia/blob/60b6d8eb133ecdc211796c714d89f877566f4bd8/core/controllers/improvements_test.py#L720-L726

Can this be fixed in a separate PR as it seems out of scope for the current one?

Reply:
Update: Per offline discussion, I will be handling these cases in the same PR.